### PR TITLE
add user authentication and profile page component

### PR DESCRIPTION
**Pull request by WoweeZowee94**
Fleshed out pre-built authentication assets to create functioning authentication, user persistence, and a profile page for the current user.

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -28,10 +29,11 @@ const App = (props) => {
       <TopBar user={currentUser} />
       <Switch>
         <Route exact path="/">
-          <h2>Hello from react</h2>
+          <h2>Hello from Pastabilites</h2>
         </Route>
         <Route exact path="/users/new" component={RegistrationForm} />
         <Route exact path="/user-sessions/new" component={SignInForm} />
+        <AuthenticatedRoute  exact path="/profile" component={UserProfile} user={currentUser} />
```
Comment (ohad-porat) [inline client/src/components/App.js:36]: We want to have only one space after `AuthentucatedRoute`
```js
<AuthenticatedRoute exact path="/profile" component={UserProfile} user={currentUser} />
```

File: client/src/components/UserProfile.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react"
+
+const UserProfile = ({ user }) => {
+  return (
+    <div>
+      <h1>This is a page built specifically for a user to see</h1>
+      <h2></h2>
```
Comment (ohad-porat) [inline client/src/components/UserProfile.js:7]: Remove any unused elements

File: client/src/components/UserProfile.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    </div>
+  )
+}
```
Comment (ohad-porat) [inline client/src/components/UserProfile.js:10]: We expect complete features to have the data that we are expecting to see and not filler text. We should at the very least see some information about the user, even if in the future we plan on having something completely different on this page.

**Review by ohad-porat — CHANGES_REQUESTED** [review]
Small and very well-scoped PR! Great work, one main thing to change is having some data to be displayed on the `UserProfile` page.

**Review by WoweeZowee94 — COMMENTED** [review]
Addressed changes suggested in first pass

**Pull request merged**